ccbuilder/builder/builder.py

```diff
@@ -359,52 +365,6 @@ def build(
         additional_patches: Optional[list[Path]] = None,
         configure_flags: str | None = None,
         jobs: Optional[int] = None,
-    ) -> Path:
-        if not jobs and not self.jobs:
-            jobs = multiprocessing.cpu_count()
-        elif not jobs:
-            jobs = self.jobs
-
-        return build_and_install_compiler(
-            project=project,
-            rev=rev,
-            cache_prefix=self.cache_prefix,
-            llvm_repo=self.llvm_repo,
-            gcc_repo=self.gcc_repo,
-            patchdb=self.patchdb,
-            get_executable=get_executable,
-            jobs=jobs,
-            additional_patches=additional_patches,
-            configure_flags=configure_flags,
-            logdir=self.logdir,
-        )
-
-
-def _worker_alive(worker_indicator: Path) -> bool:
-    if worker_indicator.exists():
-        with open(worker_indicator, "r") as f:
-            worker_pid = int(f.read())
-        try:
-            # Signal to ~check if a process is alive
-            # from man 1 kill
-            # > If signal is 0, then no actual signal is sent, but error checking is still performed.
-            os.kill(worker_pid, 0)
-            return True
-        except OSError:
-            return False
-    return False
```

Comment:
This prevented all unnecessary wait locks because of an incomplete cleanup of `WORKER_ID` etc.

I think this should be part of the test before throwing an exception.